lib/rules/no-restricted-exports.js

```diff
@@ -63,6 +114,33 @@ module.exports = {
                     messageId: "restrictedNamed",
                     data: { name }
                 });
+                return;
+            }
+
+            if (name === "default") {
+                const isSourceSpecified = node.parent.source || node.parent.parent.source;
+                const specifierLocalName = node.parent.local && node.parent.local.name;
```

Comment:
We should use `astUtils.getModuleExportName` here. For example:

```js
/* eslint no-restricted-exports: ["error", {
    restrictDefaultExports: {
        defaultFrom: false,
        namedFrom: true
    }
}]*/

export { "default" } from "foo"; // false positive; this is `defaultFrom` but was interpreted as `namedFrom`
```

Reply:
Thanks for the heads up. Fixed in https://github.com/eslint/eslint/pull/16785/commits/4bf73804d94bf8443bcc868febd594f3c1253768